addOns/automation/src/main/java/org/zaproxy/addon/automation/AuthenticationData.java

```diff
@@ -99,40 +109,55 @@ public AuthenticationData() {}
 
     public AuthenticationData(Context context) {
         AuthenticationMethod authMethod = context.getAuthenticationMethod();
-        if (authMethod instanceof HttpAuthenticationMethod) {
-            HttpAuthenticationMethod httpAuthMethod = (HttpAuthenticationMethod) authMethod;
```

Comment:
https://github.com/zaproxy/zap-extensions/pull/6120#discussion_r1928288671 not that much more difficult in this case but adds unnecessary noise.

Reply:
I don't understand what you mean.

Reply:
Oh, I think I see.